### PR TITLE
[ABACus-51] Finish testing CLI. Part One. 

### PR DESCRIPTION
**Pull request by JTR-USAF**
Initial work for gigantic TEST ALL THE THINGS story [ABACus-51]. This pull request covers tests for the BaseInteractions class, updates to __init__.py's to add src to the path in certain conditions, and finally updates to the github workflow to enforce passing tests and use of a singular PYTHON_VERSION variable.

image

**NOTE:**
The target for this pull request is my OTHER story's branch that I have in review. Once that is merged I will update the target to point to main. I simply did it this way so that you wouldn't be looking at code you've already reviewed.

### CODE REVIEW

File: src/datawave_cli/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import os
+import sys
+from pathlib import Path
+
+if os.getenv('DWV_ENV') == 'development':
+    sys.path.insert(0, Path(__file__).resolve().parents[1])
```
Comment (DL-USAF) [inline src/datawave_cli/__init__.py:6]: why is this required? what does it get us?

Comment (JTR-USAF) [reply src/datawave_cli/__init__.py:6]: If you import anything from `datawave_cli` it runs the `__init__.py`. So what this does is if you have that development environment set it ensures that the `src` directory is on the python path. Meaning that python will check `src/` for any imports first before checking `site-packages`. This means that if you're working on something and change it locally you don't have to rebuild and install (or uninstall) the installed package to get it to run your local changes. 

I'm assuming this is what was causing the problems you were having with the `.base_interactions` import. You had a version installed that didn't have `base_interactions` cause you hadn't built it yet. Therefore when you tried `import datawave_cli.base_interactions` it was checking that install (located in `site-packages`) and didn't see a `base_interactions` file and errored out. So you had to use the relative import of `.base_interactions` which says, "whatever folder this file is located in, find a file named `base_interactions` in this same level".

Best practices in python, for clarity sake, is to avoid relative import with the exception of two cases. One is if it is deeply nested and you have to repeatedly navigate through like 5 sub modules for the import. The second is if it is tightly coupled code that is not expected to ever change relative location to the file, which I suppose your use does fit. I just don't like relative imports.

After typing out this response I realize that if you're importing a file and not running that file directly `__file__` is gonna end up pointing to the site-packages version of the file anyway so it's probably moot and not needed. But the idea was solid!

Comment (DL-USAF) [reply src/datawave_cli/__init__.py:6]: the addition to the toml telling it where the src is thought fixed it. Although IDK if I did that here also or just in webapp.

Comment (JTR-USAF) [reply src/datawave_cli/__init__.py:6]: The toml telling it where the src is is only for running pytest though, no?

**Review by DL-USAF — APPROVED** [review]

**Review by JS-USAF — APPROVED** [review]

**Review by DL-USAF — APPROVED** [review]
